dotCMS/src/main/java/com/dotcms/graphql/util/GraphQLUtil.java

```diff
@@ -1,11 +1,39 @@
 package com.dotcms.graphql.util;
 
+import com.dotcms.contenttype.model.field.Field;
 import com.dotcms.graphql.InterfaceType;
+import com.dotmarketing.business.APILocator;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLType;
 import java.util.Set;
 
 public class GraphQLUtil {
 
-    public static Set<String> getFieldReservedWords() {
-        return InterfaceType.RESERVED_GRAPHQL_FIELD_NAMES;
+    public static boolean isVariableGraphQLCompatible(final Field field) {
+        // first let's check if there's an inherited field with the same variable
+        if(InterfaceType.getContentletInheritedFields().containsKey(field.name())) {
+            // now let's check if the graphql types are compatible
+
+            // get inherited field's graphql type
+            GraphQLType inheritedFieldGraphQLType = InterfaceType.getContentletInheritedFields()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'inheritedFieldGraphQLType' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995822)